### PR TITLE
FIX: #Contextify currency and language #2173

### PR DESCRIPTION
**Pull request by marcosrdz**

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/6xk6m6jfx66up6arezgrhamb7m

[download apk](http://filestorage.bluewallet.io:1488/1607824054.apk)

**Comment by limpbrains** [comment]
It is WIP, right ?
After changing language you need to update dayjs settings and re-render main screen
That would be awesome

**Comment by marcosrdz** [comment]
> It is WIP, right ?
> After changing language you need to update dayjs settings and re-render main screen
> That would be awesome

Done

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/x5pmvjx3u9wx2383c79brcky04

[download apk](http://filestorage.bluewallet.io:1488/1607854824.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/p0abqxpp6vntkb21mud176arjm

[download apk](http://filestorage.bluewallet.io:1488/1607857545.apk)

File: blue_modules/storage-context.js [hunk 1]
```diff
@@ -9,6 +11,10 @@ export const BlueStorageProvider = ({ children }) => {
   const [pendingWallets, setPendingWallets] = useState([]);
   const [selectedWallet, setSelectedWallet] = useState('');
   const [walletsInitialized, setWalletsInitialized] = useState(false);
+  const [preferredFiatCurrency, _setPreferredFiatCurrency] = useState();
+  const [language, _setLanguage] = useState();
```
Comment (Overtorment) [inline blue_modules/storage-context.js:15]: bad naming

**Comment by limpbrains** [comment]
Transaction dates are still not updating

File: blue_modules/storage-context.js [hunk 2]
```diff
@@ -21,6 +27,30 @@ export const BlueStorageProvider = ({ children }) => {
...
+    getPreferredCurrency();
+  };
+
+  const getLanguage = async () => {
+    const item = await getLanguageAsyncStorage();
+    _setLanguage(item);
+  };
+
+  const setLanguage = () => {
+    getLanguage();
```
Comment (Overtorment) [inline blue_modules/storage-context.js:45]: image

**Review by Overtorment — APPROVED** [review]
`blue_modules/storage-context.js` is unreadable. either poor naming or general lack of thought put into its design.
ok for now, but you might wanna do smth about it

**Comment by marcosrdz** [comment]
> Transaction dates are still not updating

Should  be fixed now

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/h20j06uxxq44xg73gvcyt2cxv4

[download apk](http://filestorage.bluewallet.io:1488/1607879956.apk)

**Review by limpbrains — APPROVED** [review]

**Review by ncoelho — APPROVED** [review]

**Comment by GladosBlueWallet** [comment]
Unbelievable. You, [subject name here], must be the pride of [subject hometown here]!

**Pull request merged**